sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/plans/logical/DistinctKeyVisitorSuite.scala

```diff
@@ -47,9 +45,6 @@ class DistinctKeyVisitorSuite extends PlanTest {
     assert(plan.analyze.distinctKeys === distinctKeys)
   }
 
-  implicit private def productEncoder[T <: Product : TypeTag]: ExpressionEncoder[T] =
```

Comment:
Oh, is this really unused?

Reply:
I didn't find any evidence of it being used, and it was even deemed unused in the initial pr. 

However, since there are concerns, I have reverted the changes to it in the latest commit.